staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -40,10 +39,18 @@ const nsTimeFmt = "2006-01-02 15:04:05.000000000"
 // queueSetFactory implements the QueueSetFactory interface
 // queueSetFactory makes QueueSet objects.
 type queueSetFactory struct {
-	counter counter.GoRoutineCounter
-	clock   clock.PassiveClock
+	clock             clock.PassiveClock
+	promiseMakerMaker queueSetPromiseMakerMaker
 }
 
+// queueSetPromiseMaker returns a WriteOnce
+// - whose Set method is invoked with the queueSet locked, and
+// - whose Get method is invoked with the queueSet not locked.
+type queueSetPromiseMaker func(initial interface{}, doneCh <-chan struct{}, doneVal interface{}) promise.WriteOnce
+
+// queueSetPromiseMakerMaker returns the queueSetPromiseMaker to use for the given queueSet
+type queueSetPromiseMakerMaker func(*queueSet) queueSetPromiseMaker
+
```

Comment:
How about just calling them:

promiseMaker and promiseMakerFactory ?

Reply:
I put "queueSet" into the names because they are semantically specific to a queueSet.

Reply:
The words "factory" and "maker" are interchangeable here, they both mean the same thing: "FooMaker" is a func that returns a Foo, just like "FooFactory".  Using both words suggests they mean something different --- which they do not.

Reply:
> I put "queueSet" into the names because they are semantically specific to a queueSet.

They are in queueset package already. So either they are used within this package (which is then obvious they are package specific) or they are used outside and have to be used as queueset.<type>. 
We can remove queueset without loosing the package-specific nature.

> The words "factory" and "maker" are interchangeable here, they both mean the same thing: "FooMaker"

So let's change both to Factory - I'm not just not a fan of maker. Factory is the word we use everywhere.

Reply:
done

Reply:
Thanks!